extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -93,14 +61,16 @@ class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
 
   /**
    * A URL to a GeoJSON file, or inline GeoJSON.
-   *
-   * If method is called while another asynchronous method is parsing data - asynchronous method will not
-   * apply when data is parsed.
    */
   fun data(value: String) = apply {
-    ignoreParsedGeoJsonRegistry[sourceId] = true
+    // remove any events from queue before posting this task
     workerHandler.removeCallbacksAndMessages(null)
-    setProperty(PropertyValue("data", TypeUtils.wrapToValue(value)))
+    workerHandler.post {
+      mainHandler.post {
```

Comment:
Just curious, do we need `workerHandler.post` here, as we are not doing anything on worker thread here.

Reply:
yes as we need to preserve correct order of applying data
otherwise if we're parsing smth already and apply `data` - `data` will be applied immediately and overwritten by data that was parsing afterwards.

Reply:
understood.